bio_hansel/main.py

```diff
@@ -192,36 +194,36 @@ def main():
     scheme_subtype_counts = subtype_counts(scheme_fasta)
     logging.debug(args)
     subtyping_params = init_subtyping_params(args, scheme)
-    input_genomes, reads = collect_inputs(args)
-    if len(input_genomes) == 0 and len(reads) == 0:
+    input_contigs, input_reads = collect_inputs(args)
+    if len(input_contigs) == 0 and len(input_reads) == 0:
         raise Exception('No input files specified!')
     df_md = None
     if args.scheme_metadata:
         df_md = read_metadata_table(args.scheme_metadata)
     n_threads = args.threads
 
-    subtype_results = []  # type: List[Subtype]
-    dfs = []  # type: List[pd.DataFrame]
-    if len(input_genomes) > 0:
-        query_contigs_ac(subtype_results=subtype_results,
-                         dfs=dfs,
-                         input_genomes=input_genomes,
-                         scheme=scheme,
-                         scheme_name=scheme_name,
-                         subtyping_params=subtyping_params,
-                         scheme_subtype_counts=scheme_subtype_counts,
-                         n_threads=n_threads)
-    if len(reads) > 0:
-        query_reads_ac(subtype_results=subtype_results,
-                       dfs=dfs,
-                       reads=reads,
-                       scheme=scheme,
-                       scheme_name=scheme_name,
-                       subtyping_params=subtyping_params,
-                       scheme_subtype_counts=scheme_subtype_counts,
-                       n_threads=n_threads)
+    subtype_results = []  # type: List[Tuple[Subtype, pd.DataFrame]]
+    if len(input_contigs) > 0:
+        contigs_results = subtype_contigs_samples(input_genomes=input_contigs,
+                                                  scheme=scheme,
+                                                  scheme_name=scheme_name,
+                                                  subtyping_params=subtyping_params,
+                                                  scheme_subtype_counts=scheme_subtype_counts,
+                                                  n_threads=n_threads)
+        logging.info('Generated %s subtyping results from %s contigs samples', len(contigs_results), len(input_contigs))
+        subtype_results += contigs_results
```

Comment:
Is subtype_results iterable? did you mean subtype_results.append(contigs_results)?

Reply:
It's a nice thing you can do in Python where you can add 2 lists together with `+`:
```python
list1 = [1, 2, 3] 
list1 += [4, 5]
assert list1 == [1,2,3,4,5]
#or
list1 = [1, 2, 3] 
list2 = [4, 5]
list3 = list1 + list2
assert list3 == [1,2,3,4,5]
# list1 and list2 are unmodified
```
Hopefully that makes sense!

Reply:
oh ok I see, that makes sense, thanks!